flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/SimpleDataUtil.java

```diff
@@ -284,10 +303,23 @@ public static void assertTableRecords(Table table, List<Record> expected) throws
     }
   }
 
+  public static Snapshot latestSnapshot(Table table, String branch) {
```

Comment:
shouldn't this always be `table.snapshot(branch)`? Why we need to handle main branch specially?

Reply:
Technically `table.snapshot(branch)` would also yield the same results but refer to @rdblue [comment](https://github.com/apache/iceberg/pull/6651/files#r1085892442) on the Spark write PR.

For the main branch we still want to validate the behavior of specifically currentSnapshot() is maintained since that was the original check done here. This will effectively ensure that the currentSnapshotId in metadata is maintained correctly as the main branch gets written to.

Reply:
In that case, we should put what you say as documentation for the method so people know the context. Maybe we can refactor that later with Spark, but don't need to be done now.